portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,76 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+function getdataservlet(){
+    console.log('Fetching data servlet');
+    const responsePromise = fetch('/data');
+    responsePromise.then(handleResponse);
+}
+function handleResponse(response){
+   console.log('Handling response');
+   const textPromise = response.text();
+   textPromise.then(addDataServletToDom)
+}
+
+function addDataServletToDom(servlet) {
+    console.log('Adding Data Servlet to dom:' + servlet);
+    
+    const quoteContainer = document.getElementById('dataservlet-container');
+    quoteContainer.innerText = servlet;
+}
+
+function getComments(){
```

Comment:
Again, here I would consider simplifying everything to fit inside just the getComments() function by doing something like
fetch('/data').then(response => response.json()).then((strArr) => {
    document.getElementById('comments-container').innerText = strArr;
  });
This will format the array for you, so you don't have to hard code it, but if you don't like the default formatting then you are welcome to create your own.